app/views/groups/show.html.slim

```diff
@@ -64,9 +64,12 @@
         - @group.tasks.each do |task|
           .table-list
             .btn-group
-              = link_to task.title, task_path(task), class: 'btn btn-light', style: "border-right: none"
-              = link_to remove_task_group_path(task: task), method: :patch, class: 'btn btn-light', style: "border-left: none; margin-right: 5px;"  do
-                i.fa-solid.fa-xmark style=("color:gray;")
+              - if policy(@group).remove_task?
+                = link_to task.title, task_path(task), class: 'btn btn-light', style: "border-right: none"
+                = link_to remove_task_group_path(task: task), method: :patch, class: 'btn btn-light', style: "border-left: none; margin-right: 5px;"  do
+                  i.fa-solid.fa-xmark style=("color:gray;")
+              - else
+                = link_to task.title, task_path(task), class: 'btn btn-light'
```

Comment:
I would suggest solving that in css instead of adding the `link_to task` to both cases with minimal different styling.
For example you could add a class to the `.table-list` and do 
```SCSS
.table-list.group-tasks {
  .btn-group.btn:not(:last-child) {
    border-right: none;
  }
}
```

Reply:
I agree this was not the most elegant solution :+1: